warehouse/query-core/src/main/java/datawave/query/transformer/GroupingTransform.java

```diff
@@ -157,17 +158,36 @@ private Key createAttrKey(String fieldName) {
      *            an iterator source
      * @return the flushed value that is an aggregation from the source iterator
      */
-    public Iterator<Entry<Key,Document>> getGroupingIterator(final Iterator<Entry<Key,Document>> in, int max) {
+    public Iterator<Entry<Key,Document>> getGroupingIterator(final Iterator<Entry<Key,Document>> in, int groupFieldsBatchSize, YieldCallback<Key> yieldCallback) {
         
         return new Iterator<Entry<Key,Document>>() {
             
             Entry<Key,Document> next;
             
             @Override
             public boolean hasNext() {
-                for (int i = 0; i < max && in.hasNext(); i++) {
-                    Entry<Key,Document> lastEntry = in.next();
-                    GroupingTransform.this.apply(lastEntry);
+                for (int i = 0; i < groupFieldsBatchSize; i++) {
+                    if (in.hasNext()) {
+                        GroupingTransform.this.apply(in.next());
+                    } else if (yieldCallback != null && yieldCallback.hasYielded()) {
```

Comment:
If you do not have any grouping results to return, then perhaps we should let the yield stand as is.